### PR TITLE
Deploy and use ghproxy for Prow components.

### PR DESCRIPTION
**Pull request by cjwagner**
/assign @fejta

### CODE REVIEW

File: prow/cluster/ghproxy.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    app: ghproxy
+  name: ghproxy
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 50Gi
+---
+apiVersion: extensions/v1beta1
```
Comment (fejta) [inline prow/cluster/ghproxy.yaml:15]: Deployments are v1 now, can you switch that?

File: prow/cluster/cluster.yaml [hunk 2]
```diff
@@ -483,6 +487,8 @@ spec:
         - --config-path=/etc/config/config.yaml
         - --job-config-path=/etc/job-config
         - --dry-run=false
+        - --github-endpoint=http://ghproxy
```
Comment (fejta) [inline prow/cluster/cluster.yaml:490]: Can we first deploy it and then make things use it?

Comment (cjwagner) [reply prow/cluster/cluster.yaml:490]: Sure, I'll split it up. I was going to do separate PRs originally, but the GH client will fallback to using the API directly if ghproxy isn't working for any reason so I didn't think it was very risky.

File: prow/cluster/ghproxy.yaml (hunk 1 above)
Comment (cjwagner) [reply prow/cluster/ghproxy.yaml]: Done.

File: prow/cluster/cluster.yaml (hunk 2 above)
Comment (fejta) [reply prow/cluster/cluster.yaml]: In that case might be fine to just do it but either way

Comment (cjwagner) [reply prow/cluster/cluster.yaml]: Done.
Follow up: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/345

File: prow/cluster/ghproxy.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        - containerPort: 9090
+        volumeMounts:
+        - name: cache
+          mountPath: /cache
+      volumes:
+      - name: cache
+        persistentVolumeClaim:
+          claimName: ghproxy
+---
+apiVersion: v1
```
Comment (fejta) [inline prow/cluster/ghproxy.yaml:47]: apps/v1

https://github.com/kubernetes/test-infra/blob/e0b6b08e020b8ff0276edb9d8be4f5e66ad2b37d/prow/cluster/crier_deployment.yaml#L15

Comment (cjwagner) [reply prow/cluster/ghproxy.yaml:47]: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/346
ACK about L16

**Pull request merged**